### DISCUSSION TITLE
How to extract contents in htm?

### DISCUSSION THREAD

**Discussion by Xant1k** [gildas-lormeau/SingleFile#1461] [Q&A]
Downloaded .htm and can’t extract it's contents to get images.
Any ideas?

**Comment by gildas-lormeau** [comment]
I guess the simplest solution for you is to save pages as self-extracting files or ZIP files. See the "File format" section in the options page. Thus, you will be able to unzip the files produced by SingleFile and extract their images easily.

**Reply by Xant1k** [reply]
I received this file from another user over a year ago and it was created in SingleFile.

**Reply by gildas-lormeau** [reply]
Then, you could run a script in the console as a workaround.

1. Open the saved page in a tab
2. Press `F12`
3. Paste the code below in the console and type `Enter`

```js
Array.from(document.images).forEach((image, index) => { 
  const link = document.createElement("a");
  link.href = image.src;
  link.download = "image" + index;
  link.click();
});
```

**Reply by Xant1k** [reply]
> Then, you could run a script in the console as a workaround.
> 
> 1. Open the saved page in a tab
> 2. Press `F12`
> 3. Paste the code below in the console and type `Enter`
> 
> ```js
> Array.from(document.images).forEach((image, index) => { 
>   const link = document.createElement("a");
>   link.href = image.src;
>   link.download = "image" + index;
>   link.click();
> });
> ```
Thank u for your reply. However this script downloads only the first image.
